cmd/controller/uuidgenerator.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2018 OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"strings"
+
+	libudevwrapper "github.com/openebs/node-disk-manager/pkg/udev"
+	"github.com/openebs/node-disk-manager/pkg/util"
+)
+
+const (
+	models = "EphemeralDisk,Virtual_disk"
+)
+
+func (c *Controller) GenerateUuid(device *libudevwrapper.UdevDevice) string {
+	deviceDetails := device.DiskInfoFromLibudev()
+	uid := deviceDetails.Wwn + deviceDetails.Model + deviceDetails.Serial + deviceDetails.Vendor
+	idtype := deviceDetails.Type
```

Comment:
Can we give a comment for what "idtype" is here !

Reply:
Comments for `Type` is already present in that struct. If you hover over `Type` it will show description of it I think that is ok.